appnav/src/test/kotlin/io/element/android/appnav/di/MatrixClientsHolderTest.kt

```diff
@@ -81,4 +81,17 @@ class MatrixClientsHolderTest {
         matrixClientsHolder.restoreWithSavedState(savedStateMap)
         assertThat(matrixClientsHolder.getOrNull(A_SESSION_ID)).isEqualTo(fakeMatrixClient)
     }
+
+    @Test
+    fun `test ClientAuthenticationObserver emits a value and we save it`() = runTest {
```

Comment:
Maybe rename this test now that `ClientAuthenticationObserver` has been deleted.